framework/scripts/compile_commands.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
+import sys, os
+import json
+
+lines = [line.rstrip() for line in sys.stdin]
```

Comment:
Maybe a little documentation, or a `__main__.py` method that says don't run this. 
I guess it just hangs if you attempt to run this script since it reads from stdin.
I think I'd like it if you had to pass the filename to read instead. That way you could exit if it wasn't passed with a usage.